.github/scripts/toolchain/group-gcc-test-failures.sh

```diff
@@ -0,0 +1,12 @@
+#!/bin/bash
+
+source `dirname ${BASH_SOURCE[0]}`/../config.sh
+
+TAG=$1
+
+python3 `dirname ${BASH_SOURCE[0]}`/group-gcc-test-failures.py \
```

Comment:
`dirname ${BASH_SOURCE[0]}` can be extracted to a variable to avoid repeating it.

Reply:
Used it everywhere beside sourcing to keep it consistent with other scripts as I got that as a feedback: https://github.com/Windows-on-ARM-Experiments/mingw-woarm64-build/pull/184#discussion_r1732493932

Reply:
What about moving that line to \`config.sh\`? If you decide so, I'd name it as \`ROOT\_DIR\`.

Reply:
I think that would be less transparent but if you insist, I suggest I'll do this in a follow up PR, since it's more of a refactor of existing tools.

Reply:
I do not insist. I am offering my point of view what might be a better solution from long term perspective. There should be a consensus what it the best across all participants. Feel free to disaggree while providing reasonable explanation why. When there is not a clear consensus, the decision is on the PR's author as we do not have role of a maintainer on this project who would decide otherwise.

PS: "doing something in follow up PR" is generally a road to hell. If the change has a meaning in the context of the current PR, it's better to include it with the original change, if the required changes are not harming atomicity and readability of the original change, or, if it does, do the refactor enabling the better implementation of the original PR in a PR prior to the original one, get it approved and merged first, and then finish the original PR. Just saying.

Reply:
I added the `ROOT_DIR` into `config.sh`, my reasoning was as follows:

I didn't have a strong opinion before and at the second sight, it made more sense to me to add `ROOT_DIR`, because `config.sh` already introduces a lot of commonly used directories like this.

When I was looking into this originally, I thought that introducing `ROOT_DIR` would mean to refactor all our toolchain scripts to use this variable. I was hesitant to use it here because it seemed out of the scope of this PR. But I didn't realize most of the scripts use `dirname ${BASH_SOURCE[0]}` only for sourcing `config.sh` and you can't use `ROOT_DIR` there because it's not sourced yet 😄